### PR TITLE
[1.28]  releng: Update release branch jobs for 

### PR DESCRIPTION
**Pull request by ameukam**
Follow-up of:
 - https://github.com/kubernetes/sig-release/issues/2292

/sig release
/area release-eng

cc: @gracenng @kubernetes/release-engineering 

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

### CODE REVIEW

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml [hunk 1]
```diff
@@ -159,7 +159,7 @@ periodics:
       - --allow-dup
       - --bucket=k8s-release-dev
       - --registry=gcr.io/k8s-staging-ci-images
-      - --extra-version-markers=k8s-stable1
+      - --extra-version-markers=k8s-stable2
```
Comment (xmudrii) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml:162]: This rotation is not supposed to happen now. This should happen only after cutting 1.28.0 and removing 1.24 jobs.

**Review by xmudrii — CHANGES_REQUESTED** [review]
Please fix this first.
/hold

File: config/jobs/kubernetes/generated/generated.yaml [hunk 2]
```diff
@@ -1,4 +1,231 @@
...
+      - --bare
+      - --timeout=200
+      - --scenario=kubernetes_e2e
+      - --
+      - --cluster=test-eef92b583b
+      - --check-leaked-resources
+      - --provider=gce
+      - --gcp-zone=us-west1-b
+      - --gcp-node-image=gci
+      - --extract=ci/latest-1.27
```
Comment (xmudrii) [inline config/jobs/kubernetes/generated/generated.yaml:21]: Th is also supposed to be `latest-1.28` I think.

**Comment by ameukam** [comment]
@xmudrii comments addressed. 

The CI fails on tests regarding secrets. do we changed something in the job generation ?

File: releng/test_config.yaml [hunk 3]
```diff
@@ -420,11 +418,10 @@ nodeK8sVersions:
...
     prowImage: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230719-72ce785784-master
-  # TODO(1.28): Uncomment this when adding jobs for release-1.28 branch.
-  # beta:
-  #   args:
-  #   - --repo=k8s.io/kubernetes=release-1.27
-  #   prowImage: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230719-72ce785784-1.27
+  beta:
+    args:
+    - --repo=k8s.io/kubernetes=release-1.27
+    prowImage: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230719-72ce785784-1.27
```
Comment (ameukam) [inline releng/test_config.yaml:424]: @xmudrii do I need to bump this ?

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml [hunk 4]
```diff
@@ -154,7 +154,7 @@ periodics:
       - --allow-dup
       - --bucket=k8s-release-dev
       - --registry=gcr.io/k8s-staging-ci-images
-      - --extra-version-markers=k8s-stable2
+      - --extra-version-markers=k8s-stable3
```
Comment (xmudrii) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml:157]: This is not expected, it should still stay at `k8s-stable3`

File: releng/test_config.yaml (hunk 3 above)
Comment (xmudrii) [reply releng/test_config.yaml:424]: I think yes

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml (hunk 4 above)
Comment (xmudrii) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml:157]: Same for other versions

Comment (ameukam) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml]: Done for 1.25 and 1.26.

File: config/jobs/kubernetes/generated/generated.yaml (hunk 2 above)
Comment (ameukam) [reply config/jobs/kubernetes/generated/generated.yaml]: Done.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml (hunk 1 above)
Comment (ameukam) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml]: Fixed.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml [hunk 5]
```diff
@@ -0,0 +1,2200 @@
...
+      repo: cluster-api-provider-azure
+      workdir: true
+    - base_ref: master
+      org: kubernetes-sigs
+      path_alias: sigs.k8s.io/cloud-provider-azure
+      repo: cloud-provider-azure
+    labels:
+      preset-azure-anonymous-pull: "true"
+      preset-azure-capz-sa-cred: "true"
+      preset-azure-cred-only: "true"
```
Comment (saschagrunert) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml:2040]: This line is causing the unit tests to fail:
```
jobs_test.go:556: "pull-kubernetes-e2e-capz-windows" defined in "../../jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml" may not use secret "azure-cred" in "default" cluster
```

File: releng/test_config.yaml [hunk 6]
```diff
@@ -35,35 +35,34 @@ jobs:
...
+    interval: 1h
+    sigOwners: [sig-cloud-provider-gcp]
+    releaseBlocking: true
+  ci-kubernetes-e2e-gce-cos-k8sbeta-ingress:
+    interval: 1h
+    sigOwners: [sig-cloud-provider-gcp]
+    releaseBlocking: true
+  ci-kubernetes-e2e-gce-cos-k8sbeta-default:
+    args:
+    - --env=ENABLE_CACHE_MUTATION_DETECTOR=true
```
Comment (xmudrii) [inline releng/test_config.yaml:48]: Do we have some idea if this arg is still relevant? It doesn't appear in stable1 and it hasn't been added recently, so maybe we should somehow revisit this.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.25.yaml [hunk 7]
```diff
@@ -33,14 +33,14 @@ periodics:
...
-    fork-per-release-cron: 0 14-23/24 * * *
+    fork-per-release-cron: ""
     testgrid-alert-email: gke-kubernetes-accelerators-bugs@google.com
     testgrid-alert-stale-results-hours: "24"
     testgrid-dashboards: sig-release-1.25-blocking, google-gce
     testgrid-num-failures-to-alert: "6"
     testgrid-tab-name: gce-device-plugin-gpu-1.25
   cluster: k8s-infra-prow-build
-  cron: 0 8-23/24 * * *
+  cron: 0 14-23/24 * * *
```
Comment (xmudrii) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.25.yaml:43]: This run cadence change shouldn't happen now. It should only happen upon rotating jobs and that's going to happen when pushing 1.28 from k8s-beta to k8s-stable1. That said, I think we should revert changes to all affected files (`1.25.yaml` to `1.27.yaml`)

File: releng/test_config.yaml (hunk 3 above)
Comment (xmudrii) [reply releng/test_config.yaml:424]: @ameukam This is still pending

Comment (ameukam) [reply releng/test_config.yaml:424]: @xmudrii Just to understand, if I bump the branch, I can leave the tag of the image to 1.27 or add a 1.28 variant for kubekins-e2e ?

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml (hunk 5 above)
Comment (ameukam) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml:2040]: @saschagrunert This is was added by running`go run main.go --job-config $GOPATH/src/k8s.io/test-infra/config/jobs/kubernetes/sig-release --version 1.28 --go-version 1.20.6 `. Not sure what changed. Also `pull-kubernetes-e2e-capz-windows` in 1.27 has to those presets.

File: releng/test_config.yaml (hunk 6 above)
Comment (ameukam) [reply releng/test_config.yaml:48]: Still used by sig cloud provider jobs: https://github.com/kubernetes/test-infra/blob/b11b77d3d71e77c88a172a5171c6c06fc88db9f6/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L48-L49. Not sure if we want the same signal for this job.

File: releng/test_config.yaml (hunk 3 above)
Comment (xmudrii) [reply releng/test_config.yaml:424]: @ameukam I think we should build a new kubekins-e2e variant as well prior to merging this PR.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.25.yaml (hunk 7 above)
Comment (ameukam) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.25.yaml]: Dropped the commit introducing those changes.

File: releng/test_config.yaml (hunk 3 above)
Comment (ameukam) [reply releng/test_config.yaml:424]: @xmudrii PR for the new variant.

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml (hunk 5 above)
Comment (ameukam) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.28.yaml:2040]: Updated the test in https://github.com/kubernetes/test-infra/pull/30208/commits/4a0aa14f0dd9c6359ef0917a266f3a04f869d996 to allow presubmits with azure creds.

File: releng/test_config.yaml (hunk 6 above)
Comment (xmudrii) [reply releng/test_config.yaml:48]: Okay, let's leave it as it is

**Review by xmudrii — APPROVED** [review]
Thank you, @ameukam!
/lgtm
/approve
/hold cancel

**Pull request merged**